ambry-store/src/main/java/com.github.ambry.store/CompactionLog.java

```diff
@@ -301,10 +314,11 @@ static CycleLog fromBytes(DataInputStream stream, StoreKeyFactory storeKeyFactor
       CompactionDetails compactionDetails = CompactionDetails.fromBytes(stream);
       CycleLog cycleLog = new CycleLog(compactionDetails);
       cycleLog.copyStartTime = stream.readLong();
-      cycleLog.switchStartTime = stream.readLong();
+      cycleLog.commitStartTime = stream.readLong();
       cycleLog.cleanupStartTime = stream.readLong();
       cycleLog.cycleEndTime = stream.readLong();
-      cycleLog.safeToken = StoreFindToken.fromBytes(stream, storeKeyFactory);
+      cycleLog.safeToken =
+          stream.readByte() == STORE_TOKEN_PRESENT ? StoreFindToken.fromBytes(stream, storeKeyFactory) : null;
```

Comment:
Same as the discussion [here](https://github.com/linkedin/ambry/pull/537#discussion_r99874334).